airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -398,15 +417,81 @@ def consume_logs(
+                    status = self.fetch_container_logs(
+                        pod=pod, container_name=container_logs, follow=follow_logs
+                    )
+                    pod_logging_statuses.append(status)
+                else:
+                    self.log.error(
+                        "container %s whose logs were requested not found in the pod %s",
+                        container_logs,
+                        pod.metadata.name,
+                    )
+            elif isinstance(container_logs, bool):
+                # if True is provided, get logs for all the containers
+                if container_logs is True:
+                    for container_name in all_containers:
+                        status = self.fetch_container_logs(
+                            pod=pod, container_name=container_name, follow=follow_logs
+                        )
+                        pod_logging_statuses.append(status)
+                else:
+                    self.log.error(
+                        "False is not a valid value for container_logs",
+                    )
+            else:
+                # if a sequence of containers are provided, iterate for every container in the pod
+                if isinstance(container_logs, Iterable):
+                    for container in container_logs:
+                        if container in all_containers:
+                            status = self.fetch_container_logs(
+                                pod=pod, container_name=container, follow=follow_logs
+                            )
+                            pod_logging_statuses.append(status)
+                        else:
+                            self.log.error(
+                                "Container %s whose logs were requests not found in the pod %s",
+                                container,
+                                pod.metadata.name,
+                            )
+                else:
+                    self.log.error(
+                        "Invalid type %s specified for container names input parameter", type(container_logs)
+                    )
+
+        return pod_logging_statuses
+
+    def await_container_completion(self, pod: V1Pod, container_name: str) -> bool:
         """
         Waits for the given container in the given pod to be completed
 
         :param pod: pod spec that will be monitored
         :param container_name: name of the container within the pod to monitor
+        :return: if container has completed
         """
-        while not self.container_is_terminated(pod=pod, container_name=container_name):
+        while True:
+            remote_pod = self.read_pod(pod)
+            terminated = container_is_completed(remote_pod, container_name)
+            if terminated:
+                break
+            self.log.info("Waiting for container '%s' state to be completed", container_name)
             time.sleep(1)
+        return terminated
```

Comment:
Is there any case where terminated will be false? I fail to see it.

Reply:
It is not supposed to go into the false state. The method is awaiting completion of the container. ie. It will loop until the container is completed. Let me know if it is not clear enough

Reply:
I mean - why do we need to return the bool ?

Reply:
It will always be True right ?

Reply:
Yeah good point. Makes sense to revert it back to returning None.